_data-prepper/pipelines/configuration/sinks/s3.md

```diff
@@ -44,27 +44,117 @@ Use the following options when customizing the `s3` sink.
+
+## Object key configuration
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`path_prefix` | Yes | String | The S3 key prefix path to use. Accepts date-time formatting. For example, you can use `%{yyyy}/%{MM}/%{dd}/%{HH}/` to make hourly "folders" in S3. By default, events write to the root of the bucket. 
+
+
+## codec
+
+The `codec` determines how the `s3` source formats data written to each S3 object.
+
+### avro codec
+
+The `avro` codec writes events as an [Apache Avro](https://avro.apache.org/) document.
+
+Because Avro requires a schema, you may either define the schema yourself, or Data Prepper will auto-generate a schema.
+In general, you should define your own schema as it will most accurately reflect your needs.
+In cases where your data is uniform, you may be able to automatically generate a schema.
+Auto-generated schemas are based on the first event received.
+The schema will only have keys from this event, so you must have all keys present in all events for the auto-schema generation to produce a working schema.
+Auto-generated schemas make all fields nullable.
+
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`schema` | Yes | String | The Avro [schema declaration](https://avro.apache.org/docs/current/specification/#schema-declaration). Not required if `auto_schema` is set to true.
+`auto_schema` | No | Boolean | Set to true to automatically generate the Avro [schema declaration](https://avro.apache.org/docs/current/specification/#schema-declaration) from the first event.
+
+
+### csv codec
+
+The `csv` codec writes events in comma-separated value (CSV) format.
+It also supports tab-separated value (TSV) and other delimited formats.
+
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`delimiter` | No | String | The delimiter. By default, this is `,` to support CSV.
+`header` | No | String List | The header columns.
+
+ 
+### ndjson codec
+
+The `ndjson` codec writes each single line as a JSON object.
+
+The `ndjson` codec does not take any configurations.
+
+
+### json codec
+
+The `json` codec writes events in a single large JSON file.
+Each event is written into an object within a JSON array.
+
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`key_name` | No | String | The name of the key for the JSON array. By default this is `events`.
+
+
+### parquet codec
+
+The `parquet` codec writes events into a Parquet file.
+You must set the `buffer_type` to `multipart` when using Parquet.
+
+The Parquet codec writes data using the Avro schema.
+In general, you should define your own schema as it will most accurately reflect your needs.
+In cases where your data is uniform, you may be able to automatically generate a schema.
+Auto-generated schemas are based on the first event received.
+The schema will only have keys from this event, so you must have all keys present in all events for the auto-schema generation to produce a working schema.
```

Comment:
```suggestion
The schema will only have keys from this event, so you must have all keys present in all events for the automatically generate schema to produce a working schema.
```

Reply:
I'm not sure if "auto-schema generation" is quite right.

But, the suggestion places a verb ("generate") where you need a noun (because of "the").

Maybe you mean "automatically generated schema" which would give you a noun.